server/src/modules/user/user.controller.ts

```diff
@@ -112,4 +124,46 @@ export class UserController {
       throw new BadRequestException(error.response);
     }
   }
+
+  @PublicRoute()
+  @Post('/admin')
+  async createNewAdminUser(
+    @Headers() headers: HeaderDto,
+    @Body() body: SignUpRequestBodyDto,
+    @Res({ passthrough: true }) res: Response,
+  ): Promise<IUserResponse> {
+    try {
+      const xApiKey = headers['x-api-key'];
+
+      if (!xApiKey)
+        throw new UnauthorizedException('x-api-key key not provided');
+
```

Comment:
we should have super admin auth guard for this

Reply:
implemented new SuperAdminGuard for this to validate admin routes